codegen/src/test/resources/software/amazon/awssdk/codegen/poet/rules/endpoint-resolve-interceptor.java

```diff
@@ -44,15 +55,15 @@ public SdkRequest modifyRequest(Context.ModifyRequest context, ExecutionAttribut
                     endpoint = AwsEndpointProviderUtils.addHostPrefix(endpoint, hostPrefix.get());
                 }
             }
-            List<EndpointAuthScheme> authSchemes = endpoint.attribute(AwsEndpointAttribute.AUTH_SCHEMES);
+            List<EndpointAuthScheme> endpointAuthSchemes = endpoint.attribute(AwsEndpointAttribute.AUTH_SCHEMES);
             SelectedAuthScheme<?> selectedAuthScheme = executionAttributes
                 .getAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME);
-            if (authSchemes != null && selectedAuthScheme != null) {
-                selectedAuthScheme = AwsEndpointProviderUtils.mergeIntoResolvedAuthScheme(authSchemes, selectedAuthScheme);
+            if (endpointAuthSchemes != null && selectedAuthScheme != null) {
+                selectedAuthScheme = copyEndpointSignerPropertiesToAuth(endpointAuthSchemes, selectedAuthScheme);
                 executionAttributes.putAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME, selectedAuthScheme);
             }
             if (selectedAuthScheme != null) {
-                AuthSchemeUtils.setSigningParams(executionAttributes, selectedAuthScheme.authSchemeOption());
+                copySignerPropertiesToAttributes(selectedAuthScheme.authSchemeOption(), executionAttributes);
```

Comment:
I know the context of this PR is to fix codecatalyst, but the earlier PR where this logic was added to ResolveEndpointInterceptor is already merged, so commenting here.

It seems odd for this to be responsibility of Endpoint related interceptor. It only deals with the 3 properties that endpoint rules can set, but I think we may do this generically for all signer properties, if we leave any code paths where old signer will be used (like in Presigners).

What would be broken if we didn't have this right now? I'm guessing maybe S3 related tests? Could we disable those similar to other tests that were disabled? If it is just presigner, I think we can do the generic conversion for all properties in there?

Reply:
A lot breaks without this - anything that relies on the old signers. That's a lot to disable when it is already fixed here. I'd also favor leaving it in for now.

In terms of whose responsibility it is, I put it here because it is this interceptor that (currently) finalizes the selected auth scheme's properties, but is that the right place? I don't think so, because what if something else needs to change the selected auth scheme again? Why is it the endpoint interceptor that gets to say that people aren't allowed to change it anymore?

The next idea would be to give ownership to the end of the request pipeline. Why don't we do it right before we call the signer? That way we know for sure the selected auth scheme is properly reflected in the execution attributes when they get to the legacy signer.

There's a problem with that, though. There may be legacy interceptors that rely on the values being set sooner. We want the values to be available as soon as possible to minimize the chance of backwards-incompatibility with older interceptors, so that pushes us to do it every single time we set or change the selected auth scheme. That's a very leaky abstraction.

So what do I think we should do? I think we should create what I've described in the past as "derived execution attributes". That is, ones that are determined **at the time they are read** based on other execution attributes. That way, we can define the derived execution attribute as being based on the current value of the selected auth scheme execution attribute.

For example, `executionAttributes.putDerivedAttribute(SIGNING_REGION, e -> e.getAttribute(SELECTED_AUTH_SCHEME).signingRegion())`. The advantage is that we just have to do this once - when we initially set the `SELECTED_AUTH_SCHEME`. Then we know that even if `SELECTED_AUTH_SCHEME` attribute changes, that we don't have to go and update `SIGNING_REGION` attribute.

It's not a major change. If you think this is an okay idea I can do a follow-up PR to make that change?

Reply:
That's a cool idea (and I don't recall the past well). It'd be super nice to have that in the code. It'd be useful to be able to distinguish between read-only and read-write derived attributes

Reply:
>What would be broken if we didn't have this right now?

Discussed offline, generically, this would be any cases where Signer is still used. With this project, we expect all newly generated clients that have SRA auth scheme resolution, to also use the new HttpSigner and not Signer. So it is only for when customer overrides to provide their own old interface Signer (isSignerOverridden=true cases). If customer provides their own Signer impl, we don't necessarily expect them to use our signer execution attributes because they are SdkProtectedApi - but it still have practical risk of breaking someone. But more than that, there's the case (I hadn't thought about this before discussion) where customers is explicitly providing Signer and they are using our existing SdkPublicApi (to be deprecated) Signers, say they want to use sigv4a. To support these use case, we do need ability to map new `SignerProperty`s to signer ExecutionAttributes used by old Signer - beyond the 3 endpoint related here. So @millems is going to follow up (in separate PR) on his idea above to support these.